dev-container/docker-compose.yml

```diff
@@ -0,0 +1,47 @@
+version: '3.8'
+
+services:
+  postgres:
+    image: postgres:latest
+    container_name: postgres
+    environment:
+      POSTGRES_USER: candlepin
+      POSTGRES_PASSWORD: candlepin
+      POSTGRES_DB: candlepin
+      POSTGRES_HOST_AUTH_METHOD: trust
+    ports:
+      - "5432:5432"
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U candlepin -d candlepin"]
+      interval: 5s
+      timeout: 5s
+      retries: 10
+
+  candlepin:
+    image: quay.io/candlepin/candlepin:dev-latest
+    container_name: candlepin
+    environment:
+      JPA_CONFIG_HIBERNATE_DIALECT: org.hibernate.dialect.PostgreSQL92Dialect
+      JPA_CONFIG_HIBERNATE_CONNECTION_DRIVER_CLASS: org.postgresql.Driver
+      JPA_CONFIG_HIBERNATE_CONNECTION_URL: jdbc:postgresql://postgres/candlepin
+      JPA_CONFIG_HIBERNATE_CONNECTION_USERNAME: candlepin
+      JPA_CONFIG_HIBERNATE_CONNECTION_PASSWORD: candlepin
+      CANDLEPIN_AUTH_CLOUD_ENABLE: "false"
+      CANDLEPIN_AUTH_TRUSTED_ENABLE: "true"
+      CANDLEPIN_AUTH_OAUTH_ENABLE: "true"
+      CANDLEPIN_AUTH_OAUTH_CONSUMER_RSPEC_SECRET: rspec-oauth-secret
+      CANDLEPIN_DB_DATABASE_MANAGE_ON_STARTUP: Manage
+      CANDLEPIN_REFRESH_ORPHAN_ENTITY_GRACE_PERIOD: 0
+      CANDLEPIN_STANDALONE: "false"
+      MODULE_CONFIG_HOSTED_CONFIGURATION_MODULE: org.candlepin.testext.hostedtest.HostedTestModule
```

Comment:
When we build the development image we are also populating the candlepin.conf file with [default values](https://github.com/candlepin/candlepin/blob/main/containers/release.Containerfile#L98-L109) and including it in the image. So, we could remove the duplicates that are already defined in the candlepin.conf file and only define the environment variables that differ or are not include in the default configurations.

Environment variables to include in the docker compose file:
- CANDLEPIN_AUTH_CLOUD_ENABLE
- CANDLEPIN_AUTH_TRUSTED_ENABLE
- CANDLEPIN_STANDALONE
- MODULE_CONFIG_HOSTED_CONFIGURATION_MODULE

Reply:
This one is still needed:
 JPA_CONFIG_HIBERNATE_CONNECTION_URL: jdbc:postgresql://postgres/candlepin